frontend/src/scenes/startups/startupProgramLogic.ts

```diff
@@ -0,0 +1,307 @@
+        actions: [paymentEntryLogic, ['showPaymentEntryModal']],
+    }),
+    actions({
+        setFormSubmitted: (submitted: boolean) => ({ submitted }),
+        validateYCBatch: true,
+        setYCValidationState: (state: 'none' | 'validating' | 'valid' | 'invalid') => ({ state }),
+        setYCValidationError: (error: string | null) => ({ error }),
+    }),
+    reducers({
+        formSubmitted: [
+            false,
+            {
+                setFormSubmitted: (_, { submitted }) => submitted,
+            },
+        ],
+        ycValidationState: [
+            'none' as 'none' | 'validating' | 'valid' | 'invalid',
+            {
+                setYCValidationState: (_, { state }) => state,
+                validateYCBatch: () => 'validating',
+            },
+        ],
+        ycValidationError: [
+            null as string | null,
+            {
+                setYCValidationError: (_, { error }) => error,
+                validateYCBatch: () => null,
+            },
+        ],
+    }),
+    listeners(({ actions, values }) => ({
+        validateYCBatch: async () => {
+            const { yc_batch, startup_domain, organization_name } = values.startupProgram
+
+            if (!yc_batch || yc_batch === 'Earlier') {
+                actions.setYCValidationState('valid')
+                actions.setYCValidationError(null)
+                return
+            }
+
+            try {
+                const url = `https://yc-oss.github.io/api/batches/${yc_batch.toLowerCase()}.json`
+                const response = await fetch(url)
+
+                if (!response.ok) {
+                    throw new Error('Failed to validate YC batch')
+                }
+
+                const companies = await response.json()
+                const normalizedDomain = extractDomain(startup_domain)
+                const normalizedOrgName = organization_name.toLowerCase().trim()
+
+                const foundCompany = companies.find((company: any) => {
+                    if (!company.website && !company.name) {
+                        return false
+                    }
+                    const companyDomain = company.website ? extractDomain(company.website) : null
+                    const companyName = company.name?.toLowerCase().trim() || ''
+                    const domainMatch = companyDomain === normalizedDomain
+                    const nameMatch = companyName.toLowerCase() === normalizedOrgName.toLowerCase()
+
+                    return domainMatch || nameMatch
+                })
+
+                if (foundCompany) {
+                    actions.setYCValidationState('valid')
+                    actions.setYCValidationError(null)
+                } else {
+                    actions.setYCValidationState('invalid')
+                    actions.setYCValidationError(
+                        'Could not verify YC batch membership. Please provide a screenshot of your YC profile showing "using PostHog".'
```

Comment:
tbh I don't know what this means? YC profiles don't show what tools you're using.

Reply:
I initially thought this was shown on the profile but looks like it's just shown on the deal itself

<img width="1553" alt="yc-deal-posthog-using" src="https://github.com/user-attachments/assets/0d2a41d2-d57f-417e-957c-4c8ab1c180c6" />